rustls/src/crypto/signer.rs

```diff
@@ -61,6 +62,12 @@ pub trait SigningKey: Debug + Send + Sync {
 
     /// What kind of key we have.
     fn algorithm(&self) -> SignatureAlgorithm;
+
+    /// Get the RFC 5280-compliant SubjectPublicKeyInfo (SPKI) of this [`SigningKey`] if it exists.
+    fn public_key(&self) -> Result<Option<SubjectPublicKeyInfoDer>, Error> {
```

Comment:
Nit: per the ordering explained in `CONTRIBUTING.md`, I think this should go between `choose_scheme()` and `algorithm()` (here and in any impls).

Reply:
Reordered this in [b3577ab](https://github.com/rustls/rustls/pull/1954/commits/b3577abc6b66c77c0e5024adcd8248f8f70115f7)!